changelog/0037-upgrade-to-elasticsearch7.yml

```
@@ -0,0 +1,350 @@
+      - Update below settings in your public.yml file
+        ```
+        elasticsearch_version: 7.9.1
+        elasticsearch_cluster_name: es7_<cluster_name> # must start with es7_ prefix
+        elasticsearch_download_sha256: a1b43a6e29d3ca91d08366f64007ce812646e4775524214f66330d447a4c6e3c
+        remote_es2_host: <Link to your existing ES2 cluster in http://host:port format. This will be used to whitelisted in ES7 cluster for remote reindexing>
+        ```
+      - Run commcare-cloud ansible deploy on new hosts
+        ```
+        cchq env aps --limit=new_es7_nodes --branch=<git-upgrade-branch>
+        ```
+          - If you do not have buffer resources for separate ES7 cluster or if you have a monolith, you can setup ES7 on any of your existing machines other than ES2.
+          - If you have to setup ES7 exactly on the same machines as your current ES2 cluster, follow additional steps in Appendix Note 3 before proceeding to run ES2 and ES7 on same machines. This requires at least 60% free storage/memory
+          - If you do not have extra 60% free storage/memory, follow Appendix Note 4 to upgrade to ES7 which requires full downtime and is a much slower process (suitable only for very small monoliths)
+  2. Setup a private release from which to run reindex commands. Note down the release directory and the web worker where the release is setup
+      ```
+      cchq env fab setup_limited_release:keep_days=10
+      ```
+  3. Update below settings in the above release directory's localsettings.py file
+        ```
+        ELASTICSEARCH_MAJOR_VERSION = 7
+        ELASTICSEARCH_HOSTS = [
+        ...new ES7 hosts...
+        ]
+        ELASTICSEARCH_PORT = 9200 # or whichever ES7 port you are using
+       ```
+  4. Intialize HQ indices on ES7 cluster by running this management command from the release directory created above
+        ```
+        ./manage.py initialize_es_indices
+        ```
+        - You can use an Elasticsearch monitoring tool such as https://www.elastichq.org/ to make sure the indices are created with expected number of shards. The tool will be useful to check reindex progress also.
+
+  Once you have the cluster setup, if your index sizes are small you can [reindex and switchover in one go](#reindex-and-switchover-in-one-go) which is easy but requires full application downtime. If your index sizes are large you can reindex form/case indices [first without downtime and then proceed to switchover](#reindex-and-switchover-separately) with a small downtime.
+
+
+
+  ### Reindex and Switchover in one go
+
+  If your indices are smaller than 20 GB each, the reindex and switchover phases can be combined to simplify the upgrade process. 
+
+  This requires downtime. To estimate the downtime required or to find any bugs in your ES data you may dry-run just the step 4 of below before the actual upgrade. (See Appendix Note 2, to reset the cluster before actual upgrade)
```

Wording change to get away from 'bugs' and rather refer to them as "data issues"

```suggestion
  This requires downtime. To estimate the downtime required and to surface any data issues you should run step 4 below in `dry-run` mode before the actual upgrade. (See Appendix Note 2, to reset the cluster before actual upgrade)
```

How do you do the dry run? I don't see any options in the management command to do a dry run and there aren't any more details here about how to do it.

---

> How do you do the dry run? I don't see any options in the management command to do a dry run and there aren't any more details here about how to do it.

It's here https://github.com/dimagi/commcare-cloud/pull/4219/files#diff-57488f70639428ebf4237bb113205510R93